pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -100,6 +102,17 @@ func NewCollector(cfg *Config, statesInformer statesinformer.StatesInformer, met
 	if c.config == nil {
 		c.config = NewDefaultConfig()
 	}
+	if ready := initNvml(); ready {
+		// init gpu device.
+		c.context.gpuDeviceManager = &gpuDeviceManager{}
```

Comment:
Introducing haveGPU makes the code a little confusing.  It is recommended to remove haveGPU. 

And it is recommended to abstract GPUDeviceManager into an interface. And try to hide the initialization and destruction of libnvml inside GPUDeviceManager. For example, when constructing GPUDeviceManager, if  initialize nvml successfully, return a  GPUDeviceManager implementation, otherwise return dummy GPUDeviceManager.

This simplifies the logic of checking whether the gpuDeviceManager is valid when querying GPU data.